### PR TITLE
Matrix44, PaintFilterCanvas, Fix Canvas(Bitmap) memory leak

### PR DESCRIPTION
**Pull request by igordmn**

### CODE REVIEW

File: src/main/cc/interop.cc [hunk 1]
```diff
@@ -325,6 +325,35 @@ namespace skija {
...
+            onFilterId = env->GetMethodID(local, "onFilter", "(J)Z");
+        }
+
+        void onUnload(JNIEnv* env) {
+        }
+
+        bool onFilter(jobject obj, SkPaint& paint) {
+            JNIEnv *env;
+            _vm->AttachCurrentThread((void **) &env, NULL);
+            jboolean result = env->CallBooleanMethod(obj, onFilterId, &paint);
```
Comment (olonho) [inline src/main/cc/interop.cc:344]: reinterpret_cast<jlong>(&paint)

File: src/main/cc/interop.cc [hunk 2]
```diff
@@ -325,6 +325,35 @@ namespace skija {
...
+            _vm->DetachCurrentThread();
+            return result;
+        }
+
+        void dispose(jobject obj) {
+            JNIEnv *env;
+            _vm->AttachCurrentThread((void **) &env, NULL);
+            env->DeleteGlobalRef(obj);
```
Comment (olonho) [inline src/main/cc/interop.cc:352]: Who creates this global reference?

File: src/main/java/org/jetbrains/skija/Matrix44.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.Data;
+import org.jetbrains.annotations.*;
+
+@Data
+public class Matrix44 {
+    @ApiStatus.Internal
+    public final float[] _mat;
+
+    public Matrix44(float... mat) {
+        assert mat.length == 16 : "Expected 16 elements, got " + mat == null ? null : mat.length;
```
Comment (olonho) [inline src/main/java/org/jetbrains/skija/Matrix44.java:12]: What if we run without `-ea`? Better throw `InvalidArgumentException`, IMO.

File: src/main/cc/interop.cc (hunk 2 above)
Comment (igordmn) [reply src/main/cc/interop.cc:356]: Was in PointFilterCanvas.cpp, attachToJava()

Moved here for symmetry.

File: src/main/java/org/jetbrains/skija/Matrix44.java (hunk 3 above)
Comment (igordmn) [reply src/main/java/org/jetbrains/skija/Matrix44.java:12]: I copy-pasted Matrix33 :).

assert is used in all of skija.

@tonsky, do you have opinion - what should be used: "assert" or exceptions?
In java "assert" is prettier in my opinion :), but don't work without -ea, so I would also choose exceptions.

Comment (tonsky) [reply src/main/java/org/jetbrains/skija/Matrix44.java:12]: That’s the idea. You test your app with -ea, then run it without it, not paying for asserts

**Comment by tonsky** [comment]
Pushed as fc1757bd0d31cf02481da1671a17ba51d9451fa0

**Pull request closed without merging**